From fe487b3336f8279d8b74eff05c5ebfe6e01281f0 Mon Sep 17 00:00:00 2001
From: Nicolas Lalevee <hibou@apache.org>
Date: Tue, 13 May 2014 16:52:49 +0000
Subject: [PATCH] IVY-1471 : remove the catch of the exception which makes the
 class not load if the optional jsch-agent jar is missing

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@1594279 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES.txt                                                  | 1 +
 src/java/org/apache/ivy/plugins/repository/ssh/SshCache.java | 3 +--
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CHANGES.txt b/CHANGES.txt
index b59693ef9..75c262dc8 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -148,6 +148,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 
 - FIX: impossible to get artifacts when data has not been loaded. (IVY-1399) (Thanks to David Turner)
 - FIX: regression introduced by IVY-1457, dependency management wasn't properly handled introducing lots of resolution failures
+- FIX: The SSH resolvers fails if the un-required jsch jar is missing (IVY-1471)
 
    2.4.0-rc1
 =====================================
diff --git a/src/java/org/apache/ivy/plugins/repository/ssh/SshCache.java b/src/java/org/apache/ivy/plugins/repository/ssh/SshCache.java
index f28a6a650..095ec610c 100644
--- a/src/java/org/apache/ivy/plugins/repository/ssh/SshCache.java
+++ b/src/java/org/apache/ivy/plugins/repository/ssh/SshCache.java
@@ -39,7 +39,6 @@
 import com.jcraft.jsch.Session;
 import com.jcraft.jsch.UIKeyboardInteractive;
 import com.jcraft.jsch.UserInfo;
-import com.jcraft.jsch.agentproxy.AgentProxyException;
 import com.jcraft.jsch.agentproxy.Connector;
 import com.jcraft.jsch.agentproxy.ConnectorFactory;
 import com.jcraft.jsch.agentproxy.RemoteIdentityRepository;
@@ -302,7 +301,7 @@ private boolean attemptAgentUse(JSch jsch) {
             Connector con = ConnectorFactory.getDefault().createConnector();
             jsch.setIdentityRepository(new RemoteIdentityRepository(con));
             return true;
-        } catch (AgentProxyException e) {
+        } catch (Exception e) {
             Message.verbose(":: SSH :: Failure connecting to agent :: " + e.toString());
             return false;
         }
